core/vm/evm.go

```diff
@@ -161,6 +162,9 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, state IntraBlockState, chain
 func (evm *EVM) Reset(txCtx TxContext, ibs IntraBlockState) {
 	evm.txContext = txCtx
 	evm.intraBlockState = ibs
+
+	// ensure the evm is reset to be used again
+	atomic.StoreInt32(&evm.abort, 0)
```

Comment:
1. Please check that `vm.CVMAdapter` does support reset also
2. I will work on similar method - have similar to reset evm "between blocks"